src/common/redux/reducers/keyboard.ts

```diff
@@ -22,13 +22,26 @@ export function keyboardReducer(
     ): KeyboardState {
     switch (action.type) {
         case keyboardActions.setShortcuts.ID:
-            return Object.assign({}, state, {
-                shortcuts: action.payload.shortcuts,
-            } as KeyboardState);
+            return (
+                {
+                    ...state,
+                    ...{
+                        shortcuts: action.payload.shortcuts,
+                    },
+                }
+            );
         case keyboardActions.showShortcuts.ID:
-            return Object.assign({}, state);
+            return (
+                {
+                    ...state,
```

Comment:
it may be useless, right ?

Reply:
There are several issues of this type in this PR. I propose we fix them in a second pass (subsequent PR).